src/VisualStudio/Core/Test/Venus/DocumentService_IntegrationTests.vb

```diff
@@ -0,0 +1,707 @@
+            Public LastFactory As ITableEntriesSnapshotFactory
+
+            Public Property IsStable As Boolean Implements ITableDataSink.IsStable
+
+            Public Sub AddFactory(newFactory As ITableEntriesSnapshotFactory, Optional removeAllFactories As Boolean = False) Implements ITableDataSink.AddFactory
+                LastFactory = newFactory
+            End Sub
+
+            Public Sub FactorySnapshotChanged(factory As ITableEntriesSnapshotFactory) Implements ITableDataSink.FactorySnapshotChanged
+                LastFactory = factory
+            End Sub
+
+#Region "Not Implemented"
+            Public Sub AddEntries(newEntries As IReadOnlyList(Of ITableEntry), Optional removeAllEntries As Boolean = False) Implements ITableDataSink.AddEntries
+                Throw New NotImplementedException()
+            End Sub
+
+            Public Sub RemoveEntries(oldEntries As IReadOnlyList(Of ITableEntry)) Implements ITableDataSink.RemoveEntries
+                Throw New NotImplementedException()
+            End Sub
+
+            Public Sub ReplaceEntries(oldEntries As IReadOnlyList(Of ITableEntry), newEntries As IReadOnlyList(Of ITableEntry)) Implements ITableDataSink.ReplaceEntries
+                Throw New NotImplementedException()
+            End Sub
+
+            Public Sub RemoveAllEntries() Implements ITableDataSink.RemoveAllEntries
+                Throw New NotImplementedException()
+            End Sub
+
+            Public Sub AddSnapshot(newSnapshot As ITableEntriesSnapshot, Optional removeAllSnapshots As Boolean = False) Implements ITableDataSink.AddSnapshot
+                Throw New NotImplementedException()
+            End Sub
+
+            Public Sub RemoveSnapshot(oldSnapshot As ITableEntriesSnapshot) Implements ITableDataSink.RemoveSnapshot
+                Throw New NotImplementedException()
+            End Sub
+
+            Public Sub RemoveAllSnapshots() Implements ITableDataSink.RemoveAllSnapshots
+                Throw New NotImplementedException()
+            End Sub
+
+            Public Sub ReplaceSnapshot(oldSnapshot As ITableEntriesSnapshot, newSnapshot As ITableEntriesSnapshot) Implements ITableDataSink.ReplaceSnapshot
+                Throw New NotImplementedException()
+            End Sub
+
+            Public Sub RemoveFactory(oldFactory As ITableEntriesSnapshotFactory) Implements ITableDataSink.RemoveFactory
+                Throw New NotImplementedException()
+            End Sub
+
+            Public Sub ReplaceFactory(oldFactory As ITableEntriesSnapshotFactory, newFactory As ITableEntriesSnapshotFactory) Implements ITableDataSink.ReplaceFactory
+                Throw New NotImplementedException()
+            End Sub
+
+            Public Sub RemoveAllFactories() Implements ITableDataSink.RemoveAllFactories
+                Throw New NotImplementedException()
+            End Sub
+#End Region
+        End Class
+
+        Private Class TableManager
+            Implements ITableManager
```

Comment:
Can we use Moq here?

Reply:
:trollface:

Reply:
could be, but I will see that in different PR - https://github.com/dotnet/roslyn/issues/30971